### PR TITLE
General assembly: excuse passives (fixes #417)

### LINKED ISSUE #417: Do not require passive students to join general assemblies
Add them to the excused users list by default.

Otherwise, they have a missing attendance after they become active again.

### PR DESCRIPTION
**Pull request by Trigary**

### CODE REVIEW

File: app/Models/GeneralAssemblies/GeneralAssembly.php [hunk 1]
```diff
@@ -218,4 +220,23 @@ public static function isTemporaryPasscode(string $value): bool
...
+     */
+    protected static function boot(): void
+    {
+        parent::boot();
+
+        static::created(function ($general_assembly) {
+            // Excuse passive students. They are excused at the creation of the general assembly (as opposed to when it
+            // is opened) because the list of excused students is displayed as soon as the general assembly is created.
+            foreach (Semester::current()->usersWithStatus(SemesterStatus::PASSIVE)->get() as $user) {
+                $general_assembly->excusedUsers()->attach($user);
```
Comment (kdmnk) [inline app/Models/GeneralAssemblies/GeneralAssembly.php]: I believe the attach can take a collection of users in once, so you don't need the foreach.

**Review by kdmnk — CHANGES_REQUESTED** [review]
Nice! We have some test cases for general assemblies. Can you add one to test this behaviour? Add 3 users, 1 active 2 passive, make a general assembly and check the excusedUsers if its set properly.

If you want, we could also make some improvements with the excused users here:
The table where we collect them are called "general_assembly_user" but the naming should conclude something about exclusion. Make a migration to rename it.
And another thing came into my mind is that we should add a comment/reason for the exclusion, especially if we start adding users automatically. We have some examples for pivot attributes (roles, semester status) but a simple value is enough here. You can check https://laravel.com/docs/10.x/eloquent-relationships#retrieving-intermediate-table-columns. 
Then you also need to get the value from the UI. 
Or you can open a new issue for these.

**Comment by Trigary** [comment]
Thanks, I'll make the changes you mentioned.

I just saw #300 (which mentions that observers should be preferred), therefore I just pushed a commit that makes the necessary changes. Let me know if I misunderstood something and should undo the commit.

**Comment by kdmnk** [comment]
No it's good, I just did not want to say it as we you only used the "created" event here. Observers are best if you handle a lot of events there but at least now we are consistent.

**Comment by Trigary** [comment]
I made the requested changes :)

I made two separate database migrations, because I figured a table rename is disruptive enough to warrant its own migration file. Let me know if I should combine the migrations instead.

The way the localization is handled for the `comment` column is sub-ideal: the backend fills the database with a localized string based on the locale of the user who creates the general assembly. Instead of storing a localized string, we could add some sort of flag column that determines whether the contents of the `comment` column should be displayed as-is or if it should be used as a localization key. Let me know if the current solution is adequate or if I should do something differently.

**Review by kdmnk — APPROVED** [review]
LGTM. You can store the localization key and translate when you display it. It won't mess up custom values.

**Pull request merged**